src/Transports.Subscriptions.Abstractions/Subscription.cs

```diff
@@ -56,7 +57,31 @@ public void OnCompleted()
             _unsubscribe = null;
         }
 
-        public void OnError(Exception error) => throw new NotImplementedException();
+        /// <summary>
+        /// Handles errors that are raised from the source, wrapping the error
+        /// in an <see cref="ExecutionResult"/> instance and sending it to the client.
+        /// </summary>
+        public void OnError(Exception error)
+        {
+            _logger.LogDebug("Subscription: {subscriptionId} got error", Id);
+
+            // exceptions should already be wrapped by the GraphQL engine
+            if (error is not ExecutionError executionError)
+            {
+                // but in the unlikely event that an unhandled exception delegate throws an exception,
+                // or for any other reason it is not an ExecutionError instance, wrap the error
+                executionError = new UnhandledError($"Unhandled error of type {error?.GetType().Name}", error!);
+            }
+
+            // pass along the error as an execution result instance
+            OnNext(new ExecutionResult { Errors = new ExecutionErrors { executionError } });
+
+            // Optionally we can disconnect the client by calling OnComplete() here
```

Comment:
Is it a loss of backward compatibility?

Reply:
Previously it did not terminate the connection after an error. That is why the code here is commented out. I believe, after reviewing MS recommendations on IObservable, that it should. It will can result in different behavior depending on user code.

If OnError is not called by the observable then there is no difference. If it is, and if they are using System.Reactive which will discard packets after OnError, then this change actually improves support because it informs the client that no more packets will be received. 

And keep in mind that this library used System.Reactive previously so it would not have been possible to send additional packets after OnError regardless of what the user code did.

Reply:
> after reviewing MS recommendations on IObservable

OK, but keep in mind that all that MS stuff around observables is an implementation details. They can give any recommendations, but these recommendations have a lower priority compared with the GraphQL design/protocols. The behavior of a specific implementation should not dictate the behavior of the top-level design.

Reply:
See:
- #755